pkgs/applications/window-managers/i3/bumblebee-status/plugins.nix

```diff
@@ -0,0 +1,385 @@
+{ lib
+, pkgs
+, python3Packages
+, ...
+}:
+with pkgs;
+with python3Packages; let
+  mkPluginCfg =
+    { name
+    , requires ? [ ]
+    , requiresUnpackaged ? [ ]
+    , optional ? [ ]
+    , warnings ? [ ]
+    ,
+    }:
+    let
+      warningsList =
+        let
+          warn = dep: ''Plugin "${name}": Dependency `${dep}` is not known to be contained in Nixpkgs, so plugin might fail at runtime.'';
+          requiresUnpackagedWarnings = map warn requiresUnpackaged;
+        in
+        warnings ++ requiresUnpackagedWarnings;
+    in
+    rec {
+      inherit name requires requiresUnpackaged optional;
+      warnings = warningsList;
+      hasWarnings = warnings != [ ];
+      warningsToString = lib.strings.concatStringsSep " " warnings;
```

Comment:
Adding these two attributes, which are derived from `warnings` in a simple way, seems strange to me. I'd rather create variables for them wherever you actually *use* them. It turns out the usage actually is in `default.nix`, so you split up the logic for processing warnings into two files. I'd suggest in `plugins.nix` you just define `warnings`, and all further computation should be done in the block for `propagatedBuildInputs` in `default.nix`, using variables as you like.